examples/advanced/sklearn-svm/job_configs/sklearn_svm_base/app/custom/svm_learner.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+from sklearn.metrics import roc_auc_score
+from sklearn.svm import SVC
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_opt.sklearn.sklearner import SKLearner
+
+
+class SVMLearner(SKLearner):
+    def __init__(
+        self,
+        data_path: str,
+        train_start: int,
+        train_end: int,
+        valid_start: int,
+        valid_end: int,
+    ):
+        super().__init__(data_path, train_start, train_end, valid_start, valid_end)
+        self.train_data = None
+        self.valid_data = None
+        self.n_samples = None
+        self.svm = None
+        self.kernel = None
+        self.params = {}
+
+    def initialize(self, fl_ctx: FLContext):
+        self.fl_ctx = fl_ctx
+        data = self.load_data()
+        self.train_data = data["train"]
+        self.valid_data = data["valid"]
+        # train data size, to be used for setting
+        # NUM_STEPS_CURRENT_ROUND for potential use in aggregation
+        self.n_samples = data["train"][-1]
+        # model will be created after receiving global parameter of kernel
+
+    def train(self, curr_round: int, global_param: Optional[dict] = None) -> dict:
+        if curr_round == 0:
+            # only perform training on the first round
+            # the following rounds directly returns the retained records
```

Comment:
instead of "following rounds directly returns the retained records",
would it be better to show "warning messages" and stop the learning?
otherwise there might be people setting rounds to 100 (carelessly) and later figure out only first round is needed.

Reply:
added system_panic